### ISSUE TITLE
problems on 4.3.2 that were not on 4.2.0, maybe with image dimensions?

### ISSUE THREAD

**Issue by t4k** [samvera/serverless-iiif#116]
The serverless-iiif part of the stack might not be where my issue is, but I need a place to start.

I have been developing against a test install of this application that was spun up on version 4.2.0 and it works without issue. I'm ready to move to production and spinning up a new install of this application at 4.3.2 is giving me lots of trouble.

I'm seeing messages like _The image “https://rvmnf2goqim5qiqtpufd544upq0wyoje.lambda-url.us-west-2.on.aws/iiif/2/76832775/full/90,/0/default.jpg” cannot be displayed because it contains errors._ (in the browser) to 
```
Unable to get dimensions for 76832775 using custom function. Falling back to sharp.metadata().
``` 
and
```
TypeError: Cannot read properties of undefined (reading 'toString')
    at /var/task/cache.js:31:54
    at new Promise (<anonymous>)
    at makeCache (/var/task/cache.js:29:10)
    at handleImageRequestFunc (/var/task/index.js:70:13)
    at async handleResourceRequestFunc (/var/task/index.js:49:14)
    at async Runtime.handleRequestFunc [as handler] (/var/task/index.js:24:16)
```
(in CloudWatch)

Please let me know what additional information might be helpful in diagnosing/debugging this issue. I'm happy to provide it.

**Comment by t4k** [comment]
Absolutely. Here is a generated original TIFF and a Pyramid TIFF:

- https://caltech.box.com/s/89ymy7bryj61f1vxhit3fb7tefzz3zg6
- https://caltech.box.com/s/j7vlsnh2axbaz73d8q1tbpaj3767pd24

And here is a real-world original and pyramid:

- https://caltech.box.com/s/3p5dbdphgfgg8zti4ik5rzkcdj9fscz7
- https://caltech.box.com/s/9k1tbgc4pciwmbs71xuia7bwpezuvwbg

The Pyramid TIFFs were created with the same `vips` command as in the README with the only exception being the file extension on the output file is set to `.ptif`. (The `resolverTemplate` variable in my settings is updated to `%s.ptif` as well.)

**Comment by mbklein** [comment]
Great, thanks. I’ll let you know what I find.

**Comment by mbklein** [comment]
Good morning! I've just published v4.3.3 which I _think_ should resolve the issue. I wasn't able to reproduce your exact error, but I did find a rounding error in the resizing code. Your image is currently resolving on my 4.3.3 test deployment at 
https://m7hwdiiq7opcbpc3ernelcsdme0uzppb.lambda-url.us-east-1.on.aws/iiif/2/76832775/full/90,/0/default.jpg. I'll leave that stack running until this issue is closed to make it easier to continue to work on it if necessary.
